Server for VILA-M3 integrated to the slicer module

diazandr3s opened the pull request:
- Server working for NVCF functions to interact with 3D Slicer
- This model works from images pre loaded to NVCF

diazandr3s commented:
@mingxin-zheng could you please review and approve? :)

mingxin-zheng commented:
Hi @adiazpinto , thanks for the PR. Are you importing these from [util.py](https://github.com/Project-MONAI/VLM/blob/main/m3/demo/experts/utils.py)?
```
from utils import ChatHistory, ImageCache, M3Generator, SessionVariables
```
Some of the classes above are in that file, but some are not.

Also, it worth noting that streaming mode isn't supported at the moment. I saw you have an implementation in the server, but it is unclear to me how that would be enabled for the internal NVCF function or local server setup in the repo.

Thanks!

diazandr3s commented:
> Hi @adiazpinto , thanks for the PR. Are you importing these from [util.py](https://github.com/Project-MONAI/VLM/blob/main/m3/demo/experts/utils.py)?
> 
> ```
> from utils import ChatHistory, ImageCache, M3Generator, SessionVariables
> ```
> 
> Some of the classes above are in that file, but some are not.
> 
> Also, it worth noting that streaming mode isn't supported at the moment. I saw you have an implementation in the server, but it is unclear to me how that would be enabled for the internal NVCF function or local server setup in the repo.
> 
> Thanks!

Thanks again, @mingxin-zheng.
I've updated the PR. I've also removed the streaming part. We may need that in the future, though :)

mingxin-zheng commented on server-slicer/utils.py:
```diff
@@ -0,0 +1,1040 @@
+class ExpertVista3D():
+    """Expert model for VISTA-3D."""
+
+    NIM_VISTA3D = "https://api.nvcf.nvidia.com/v2/nvcf/pexec/functions/caf65f74-1c04-4b3c-b566-5908ee950f24"
```
Hi @diazandr3s 
This is an internal NVCF. Having it exposed in the code is not intended.

mingxin-zheng commented on server-slicer/utils.py:
```diff
@@ -0,0 +1,1040 @@
+            )
+            logger.info(f"Model {self.model_name} loaded successfully. Context length: {self.context_len}")
+        elif source == "nim":
+            self.base_url = "https://api.nvcf.nvidia.com/v2/nvcf/pexec/functions/a2dec46a-b444-45aa-a1fc-a510ca41f186"
```
Same as above. Thanks!

mingxin-zheng commented on server-slicer/utils.py:
```diff
@@ -0,0 +1,1040 @@
+
+REMOTE_URL = "https://developer.download.nvidia.com/assets/Clara/monai/samples"
+
+MODEL_CARDS = (
```
I think some variables, classes and functions in this script are exactly the same as the ones in [utils.py](https://github.com/Project-MONAI/VLM/blob/main/m3/demo/experts/utils.py)

Can we include the `utils.py` in the PYTHON path and import them to avoid duplicate code? Thanks!

diazandr3s replied:
Good point, @mingxin-zheng. We should actually unified this into a single folder? Should i move this to the demo folder?

mingxin-zheng replied:
True - maybe we should make a general util folder instead of copy-pasting snippet between folders.  Can you add a `TODO: unify this class with demo/experts/utils.py" and we can check this later?
